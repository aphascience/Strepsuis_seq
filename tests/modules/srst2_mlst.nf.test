nextflow_process {

    name "Test Process SRST2_MLST"
    script "modules/srst2_mlst.nf"
    process "SRST2_MLST"

    test("Should map reads to mlst reference") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = ["test", file("${projectDir}/tests/data/input_S999_R1_001.fastq.gz"), file("${projectDir}/tests/data/input_S999_R2_001.fastq.gz")]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
