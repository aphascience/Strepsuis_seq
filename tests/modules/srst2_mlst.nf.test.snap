{
    "Should map reads to mlst reference": {
        "content": [
            {
                "0": [
                    [
                        "test",
                        "MLST_test__mlst__Ssuis_MLSTalleles__results.txt:md5,5f3b501c2ebc5d6c8650fd9766d54ec6"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-24T11:27:49.481176292"
    }
}